bundles/persistence/org.openhab.persistence.dynamodb/java/org/openhab/persistence/dynamodb/internal/DynamoDBPersistenceService.java

```diff
@@ -275,7 +271,19 @@ private synchronized boolean createTable(DynamoDBMapper mapper, Class<?> dtoClas
             }
 
             // table found or just created, wait
-            return waitForTableToBecomeActive(tableName);
+            boolean tableActive = waitForTableToBecomeActive(tableName);
+            if (tableActive && dbConfig.isAutoExpirationEnabled()){
+                db.getDynamoClient().updateTimeToLive(
+                        new UpdateTimeToLiveRequest()
+                                .withTableName(tableName)
+                                .withTimeToLiveSpecification(
+                                        new TimeToLiveSpecification()
+                                        .withEnabled(true)
+                                        .withAttributeName(DynamoDBItem.ATTRIBUTE_NAME_EXPIRATION)
+                                )
+                );
+            }
+            return tableActive;
```

Comment:
A question with implementation details -- is it possible that table is not active since auto-expiration parameter was just modified?

Reply:
Please see this comment @mindstorms6

Reply:
Please see this comment @mindstorms6